In [21]:
#Airbnb Project - 2022

#Part 1: Data preparation and visualization 

#import all the necessary libraries 
import pandas as pd
import numpy as np 
import matplotlib
import matplotlib.pyplot  as plt
%matplotlib inline
import seaborn as sns 

#load the file 
df = pd.read_csv(r'C:\Users\arlin\OneDrive\Υπολογιστής\FALL 2022\business analytics_fall 22\project\listings_CPH.csv')

In [5]:
#show dimensionality of the dataset 
df.shape

(13815, 18)

This dataset has 13815 observations and 18 features. To proceed with the analysis, data wrangling should be done. 

In [6]:
df.head()

id                                               name  host_id  \
0   6983                               Copenhagen 'N Livin'    16774   
1  26057                Lovely house - most attractive area   109777   
2  26473          City Centre Townhouse Sleeps 1-10 persons   112210   
3  29118                    Best Location in Cool Istedgade   125230   
4  31094  Beautiful, spacious, central, renovated Penthouse   129976   

  host_name  neighbourhood_group              neighbourhood   latitude  \
0     Simon                  NaN                    Nrrebro  55.686410   
1      Kari                  NaN                   Indre By  55.693070   
2     Julia                  NaN                   Indre By  55.676020   
3      Nana                  NaN  Vesterbro-Kongens Enghave  55.670230   
4      Ebbe                  NaN  Vesterbro-Kongens Enghave  55.666602   

   longitude        room_type  price  minimum_nights  number_of_reviews  \
0  12.547410  Entire home/apt    898               3                172   
1  12.576490  Entire home/apt   2600               4                 59   
2  12.575400  Entire home/apt   3250               3                300   
3  12.555040  Entire home/apt    725               7                 24   
4  12.555283  Entire home/apt   1954               3                 19   

  last_review  reviews_per_month  calculated_host_listings_count  \
0  2022-06-21               1.08                               1   
1  2022-08-09               0.55                               1   
2  2022-09-10               2.06                               3   
3  2022-08-04               0.16                               1   
4  2022-08-22               0.13                               1   

   availability_365  number_of_reviews_ltm  license  
0                 0                      4      NaN  
1               303                      8      NaN  
2                56                      7      NaN  
3                59                      2      NaN  
4                 0                      2      NaN

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13815 entries, 0 to 13814
Data columns (total 18 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              13815 non-null  int64  
 1   name                            13815 non-null  object 
 2   host_id                         13815 non-null  int64  
 3   host_name                       13814 non-null  object 
 4   neighbourhood_group             0 non-null      float64
 5   neighbourhood                   13815 non-null  object 
 6   latitude                        13815 non-null  float64
 7   longitude                       13815 non-null  float64
 8   room_type                       13815 non-null  object 
 9   price                           13815 non-null  int64  
 10  minimum_nights                  13815 non-null  int64  
 11  number_of_reviews               13815 non-null  int64  
 12  last_review                     

Two the columns neighbourhood_group and licence are empty. They should be dropped as there are no recordings associated with them. 

In [27]:
#drop columns with no recordings 
new_df = df.drop(['neighbourhood_group', 'license'],axis=1)

In [28]:
#set the index
new_df.set_index('id')

name  \
id                                                                      
6983                                             Copenhagen 'N Livin'   
26057                             Lovely house - most attractive area   
26473                       City Centre Townhouse Sleeps 1-10 persons   
29118                                 Best Location in Cool Istedgade   
31094               Beautiful, spacious, central, renovated Penthouse   
...                                                               ...   
653494030951422457  Top-floor Villa Apartment in the Heart of Hell...   
647809303952891559  Dejligt lille hus med flere hyggekroge ude og ...   
650819220455514228  Lejlighed i Storkøbenhavn. 13 minutter fra cen...   
658612163688161695     Cosy apartment with a great view in Copenhagen   
648436253362373119     Big Bedroom connected with a large living room   

                      host_id host_name              neighbourhood   latitude  \
id                                                                              
6983                    16774     Simon                    Nrrebro  55.686410   
26057                  109777      Kari                   Indre By  55.693070   
26473                  112210     Julia                   Indre By  55.676020   
29118                  125230      Nana  Vesterbro-Kongens Enghave  55.670230   
31094                  129976      Ebbe  Vesterbro-Kongens Enghave  55.666602   
...                       ...       ...                        ...        ...   
653494030951422457  465189427   Martine                    sterbro  55.732840   
647809303952891559    4862421  Kathrine                      Valby  55.667782   
650819220455514228  134647873       Tim                      Valby  55.659536   
658612163688161695  256903668       Sam               Brnshj-Husum  55.730971   
648436253362373119  141288846     Tanja                 Bispebjerg  55.731420   

                    longitude        room_type  price  minimum_nights  \
id                                                                      
6983                12.547410  Entire home/apt    898               3   
26057               12.576490  Entire home/apt   2600               4   
26473               12.575400  Entire home/apt   3250               3   
29118               12.555040  Entire home/apt    725               7   
31094               12.555283  Entire home/apt   1954               3   
...                       ...              ...    ...             ...   
653494030951422457  12.572370  Entire home/apt   1250              14   
647809303952891559  12.463285  Entire home/apt    450               3   
650819220455514228  12.474726  Entire home/apt    400               1   
658612163688161695  12.487993  Entire home/apt    850               1   
648436253362373119  12.521770     Private room    420               2   

                    number_of_reviews last_review  reviews_per_month  \
id                                                                     
6983                              172  2022-06-21               1.08   
26057                              59  2022-08-09               0.55   
26473                             300  2022-09-10               2.06   
29118                              24  2022-08-04               0.16   
31094                              19  2022-08-22               0.13   
...                               ...         ...                ...   
653494030951422457                  0         NaN                NaN   
647809303952891559                  3  2022-07-21               1.01   
650819220455514228                  3  2022-08-14               1.27   
658612163688161695                  0         NaN                NaN   
648436253362373119                  1  2022-07-07               0.37   

                    calculated_host_listings_count  availability_365  \
id                                                                     
6983                          

In [29]:
#sort the dataset on the price column 
new_df.sort_values(by='price', ascending=True)

id                                               name    host_id  \
7786  46325896                  citizenM Copenhagen Radhuspladsen  374968389   
7183  43078527                          First Hotel Kong Frederik  316080552   
6545  39000504                       Cosy room in fabulous condo!   25580883   
6876  41285362              Copenhagen loft apartment at Nørrebro  116485169   
7986  48652028              Room for rent in shared accommodation  225782385   
...        ...                                                ...        ...   
7496  44226456            Grøn landsbyidyl i hjertet af København   70535897   
7494  44224836  Cozy and bright 2-room apartment with big balcony   83120126   
2902  16397010                         Cosy and spacey apartment!   33025844   
5473  33108071             not available for bookings due to sale   44675877   
2139  11339117    “My Oasis”... a Cosy room in the Harte off Cph.   59346390   

                      host_name              neighbourhood   latitude  \
7786                   CitizenM                   Indre By  55.677323   
7183  First Hotel Kong Frederik                   Indre By  55.677570   
6545                  Annabella                    Nrrebro  55.689410   
6876                     Olivia                 Bispebjerg  55.699910   
7986                     Nitesh                 Bispebjerg  55.703160   
...                         ...                        ...        ...   
7496                      Nanna                    sterbro  55.702290   
7494                     Martin                 Bispebjerg  55.707100   
2902                      Jacob              Frederiksberg  55.689210   
5473                  Christian  Vesterbro-Kongens Enghave  55.657810   
2139                       June                Amager Vest  55.662210   

      longitude        room_type  price  minimum_nights  number_of_reviews  \
7786  12.565638       Hotel room      0               1                  0   
7183  12.567790       Hotel room      0               1                  0   
6545  12.562600     Private room     79               1                 28   
6876  12.536020  Entire home/apt    113               1                  0   
7986  12.534220     Private room    120               1                  0   
...         ...              ...    ...             ...                ...   
7496  12.574260  Entire home/apt  41800               3                  5   
7494  12.525490  Entire home/apt  44000             500                  1   
2902  12.518130  Entire home/apt  58000              14                 16   
5473  12.561920  Entire home/apt  62000              27                  0   
2139  12.579380     Private room  64900               1                 47   

     last_review  reviews_per_month  calculated_host_listings_count  \
7786         NaN                NaN                               1   
7183         NaN                NaN                               1   
6545  2022-09-15               0.77                               2   
6876         NaN                NaN                               1   
7986         NaN                NaN                               1   
...          ...                ...                             ...   
7496  2021-08-16               0.19                               2   
7494  2020-08-14               0.04                               1   
2902  2020-03-15               0.23                               1   
5473         NaN                NaN                               1   
2139  2021-11-08               0.60                               1   

      availability_365  number_of_reviews_ltm  
7786                 0                      0  
7183                 0                      0  
6545                38                     22  
6876                 0                      0  
7986                 0                      0  
...                ...                    ...  
7496               177                      0  
